summarization/summarizer_main.py

```diff
@@ -33,6 +35,9 @@ def __init__(
             text.embedding for text in self.normal_text_contents
         ]
 
+        # fetch embeddings for embedded content types
+        self._fetch_and_set_stamp_descriptors_dict_for_embedded_content()
```

Comment:
Can we load these embeddings only in the case that we have embedded content in the STAMP page? Also, for such embeddings, it would be great if we could ensure that they are loaded just once per run and not once every web article or STAMP page.

Reply:
That makes sense! thanks for pointing this out!

Reply:
Updated to load it only if there are embedded content!